lib/AboutPreferences.jsm

```diff
@@ -279,10 +283,18 @@ this.AboutPreferences = class AboutPreferences {
       contentDiscoveryButton.textContent = formatString("prefs_content_discovery_button");
       createAppend("hbox", discoveryGroup)
         .appendChild(contentDiscoveryButton)
-        .addEventListener("click", () => {
-          discoveryGroup.style.display = "none";
-          contentsGroup.style.visibility = "visible";
-          Services.prefs.clearUserPref(DISCOVERY_STREAM_CONFIG_PREF_NAME);
+        .addEventListener("click", async () => {
+          const activeExperiments = await PreferenceExperiments.getAllActive();
+          const experiment = activeExperiments.find(exp => exp.preferenceName === DISCOVERY_STREAM_CONFIG_PREF_NAME);
+          if (experiment) {
```

Comment:
Hmm.... this has an odd behavior of discovery stream can be enabled according to `DiscoveryStream.config.enabled` but the button only does something if the experiment is active.

If we condition showing opt-out based on an active experiment, it makes it hard to test by just flipping the pref. But depending on how long these async calls take, waiting on that to determine what to show might be problematic too.

Maybe we can just conditionally stop the experiment and unconditionally (and immediately before `await`ing) hide. If the tester manually changed the pref without being in the experiment, the pref will be unchanged, but at least can confirm the before/after UI states.